src/code_104940_len_dc0.c

```diff
@@ -1,37 +1,336 @@
+
+    func_802E3A00(entity);
+    temp_f20 = entity->position.y;
+    entity->position.y = temp_f20 + ((f64)sin_rad((temp->unk_18 * 6.28318f) / 360.0f));
+    temp->unk_18 += 60.0f;
+    if (temp->unk_18 >= 360.0f) {
+        temp->unk_18 = 0.0f;
+        entity->position.y = temp->unk_14;
+        func_8010FD68(entity);
+    }
+}
+
+void func_802E3898(Entity* entity) {
+    struct802E3650* temp;
+
+    entity_init_Hammer23Block_normal(entity);
+    temp = (struct802E3650*)entity->dataBuf;
+    temp->unk_04 = 0xA;
+    temp->unk_06 = 0x7B;
+    temp->unk_0A = 0xFFFF;
+}
+
+extern s32 D_802E9E54;
+extern StaticEntityData D_802EA07C;
+
+void func_802E38D8(Entity* entity) {
+    struct802E3650* temp = (struct802E3650*)entity->dataBuf;
+    s32 phi_v1;
+    s32 phi_v0;
+
+    if (temp->unk_04 > 0) {
+        if (temp->unk_04 == 10) {
+            phi_v1 = 21;
+            phi_v0 = temp->unk_0A;
+        } else {
+            phi_v1 = 20;
+            phi_v0 = 0;
+        }
+        make_item_entity_nodelay(0x157, entity->position.x, entity->position.y + 28.0, entity->position.z, phi_v1, phi_v0);
+        temp->unk_04 -= 1;
+    }
+
+    if ((temp->unk_04 == 0) || (temp->unk_06 == 0)) {
+        temp->unk_03 = 1;
+        func_8010FBC0(get_entity_by_index(create_entity(&D_802EA07C, entity->position.x, entity->position.y, entity->position.z,
+                                          entity->rotation.y, 0x80000000)), &D_802E9E54);
+        entity->flags |= 0x20000020;
+    }
+}
+
+void func_802E3A00(Entity* entity) {
+    u16 temp_v0;
+    struct802E3650* temp = (struct802E3650*)entity->dataBuf;
+    s16 phi_return;
+
+    if (entity->unk_0A == 0xE) {
+        if (temp->unk_04 != 0xA) {
+            temp->unk_06 -= 1;
+            if (temp->unk_06 < 0) {
+                temp->unk_06 = 0;
+                temp->unk_04 = 1;
+            }
+        }
+    }
+}
+
+void func_802E3A48(Entity* entity) {
+    s32 temp_v0;
+    struct802E3650* temp = (struct802E3650*)entity->dataBuf;
+
+    if ((entity->unk_06 & 0x80) != 0) {
+        func_8010FD68(entity);
+        return;
+    }
+    func_802E3A00(entity);
+    func_802E3650(entity);
+    if (temp->unk_03 != 0) {
+        create_entity(&D_802EA07C, entity->position.x, entity->position.y, entity->position.z, entity->rotation.y, 0x80000000);
+        entity->flags |= 0x20000020;
+    }
+}
+
+void func_802E3B08(Entity* entity) {
+    s32 temp_v0;
+    u16 temp_v1;
+    struct802E3650* temp = entity->dataBuf;
+
+    if (temp->unk_0A != 0xFFFF) {
+        if (get_global_flag(temp->unk_0A) != 0) {
+            create_entity(&D_802EA07C, entity->position.x, entity->position.y, entity->position.z, entity->rotation.y, 0x80000000);
+            entity->flags |= 0x20000020;
+        }
+    }
+}
 
-INCLUDE_ASM(s32, "code_104940_len_dc0", entity_init_Hammer23Block_normal);
+extern s32 D_802E9E80;
 
-INCLUDE_ASM(s32, "code_104940_len_dc0", func_802E3714);
+#ifdef NON_MATCHING
```

Comment:
Would be good to have a comment above explaining why this doesnt match (probably rodata?)